xmldiffs

```diff
@@ -82,20 +82,28 @@ else:
         return fp
 
 def xmldiffs(file1, file2, diffargs=["-u"]):
-    tree = ET.parse(file1)
-    tmp1 = unicode_writer(NamedTemporaryFile('w'))
-    write_sorted(tmp1, tree.getroot())
-    tmp1.flush()
-
-    tree = ET.parse(file2)
-    tmp2 = unicode_writer(NamedTemporaryFile('w'))
-    write_sorted(tmp2, tree.getroot())
-    tmp2.flush()
+    try:
+        tree = ET.parse(file1)
+        tmp1 = unicode_writer(NamedTemporaryFile('w'))
+        write_sorted(tmp1, tree.getroot())
+        tmp1.flush()
+        actual = tmp1.name
+    except ET.ParseError:
+        actual = file1
```

Comment:
So in case the XML has errors, we simply fall back to a diff of the original files?

Reply:
yes!